src/Public/OpenApi.ps1

```diff
@@ -1247,47 +1792,165 @@ An component schema name.
+
+.PARAMETER MaxProperties
+If supplied, will restrict the maximum number of properties allowed in an schema.
+
+.PARAMETER Array
+If supplied, the schema will be treated as an array of objects.
+
+.PARAMETER UniqueItems
+If supplied, specify that all items in the array must be unique
+
+.PARAMETER MinItems
+If supplied, specify minimum length of an array 
+
+.PARAMETER MaxItems
+If supplied, specify maximum length of an array 
+
+.PARAMETER Xml
+If supplied, controls the XML serialization behavior
+
 
 .EXAMPLE
 New-PodeOASchemaProperty -Name 'Config' -ComponentSchema "MyConfigSchema"
 #>
 function New-PodeOASchemaProperty
 {
-    [CmdletBinding()]
+    [CmdletBinding(DefaultParameterSetName = 'Inbuilt')]
     param(
-        [Parameter()]
+        [Parameter(Mandatory = $true)]
         [string]
         $Name,
 
-        [Parameter(Mandatory=$true)]
+        [Parameter(Mandatory = $true)]
         [string]
-        $Reference,
+        $ComponentSchema, 
 
-        [Parameter()]
+        [Parameter(ParameterSetName = 'Array')] 
         [string]
         $Description,
 
+        [Parameter(ParameterSetName = 'Array')] 
+        [String]
+        $Example,
+
+        [Parameter(ParameterSetName = 'Array')]
         [switch]
-        $Array
-    )
+        $Deprecated, 
 
-    if(!(Test-PodeOAComponentSchema -Name $Reference)) {
-        throw "The OpenApi component schema doesn't exist: $($Reference)"
-    }
+        [Parameter(ParameterSetName = 'Array')]
+        [switch]
+        $Required,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [switch]
+        $Nullable, 
+
+        [Parameter(ParameterSetName = 'Array')]
+        [switch]
+        $ReadOnly,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [switch]
+        $WriteOnly,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [int]
+        $MinProperties,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [int]
+        $MaxProperties,
 
+        [Parameter(Mandatory = $true, ParameterSetName = 'Array')]
+        [switch]
+        $Array,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [switch] 
+        $UniqueItems,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [int]
+        $MinItems,
+
+        [Parameter(ParameterSetName = 'Array')]
+        [int]
+        $MaxItems, 
+
+        [hashtable[]]
+        $Xml
+    )
+    if ( !(Test-PodeOAComponentSchema -Name $ComponentSchema))
+    {
+        throw "The OpenApi component schema doesn't exist: $ComponentSchema"
+    }  
     $param = @{
-        name = $Name
-        type = 'schema'
-        schema = $Reference
-        array = $Array.IsPresent
-        description = $Description
+        name   = $Name
+        type   = 'schema' 
+        schema = $ComponentSchema 
+        meta   = @{}
     }
+    if ($PSCmdlet.ParameterSetName.ToLowerInvariant() -eq 'array')
+    {   
+        if ($Description ) { $param.description = $Sescription }
```

Comment:
`$Description` 😉

Reply:
oppsssss

Reply:
this is the issue with PowerShell nothing is telling you when you write rubbish

Reply:
I have 5 $Sescription

Reply:
fixed